src/main/java/org/springframework/data/redis/core/DefaultReactiveListOperations.java

```diff
@@ -285,7 +288,8 @@ public Mono<V> rightPopAndLeftPush(K sourceKey, K destinationKey, Duration timeo
 		Assert.isTrue(isZeroOrGreater1Second(timeout), "Duration must be either zero or greater or equal to 1 second");
 
 		return createMono(
-				connection -> connection.bRPopLPush(rawKey(sourceKey), rawKey(destinationKey), timeout).map(this::readValue));
+				connection -> connection.bRPopLPush(rawKey(sourceKey), rawKey(destinationKey), timeout)
+						.map(this::readRequiredValue));
```

Comment:
Shouldn't this be `.mapNotNull(..)` given the `Duration` parameter?

Reply:
The previous variant was `mapNotNull` that effectively filters `null` values turning `RedisElementReader` into something that has an ability of filtering. That's why we chose to use `readRequiredValue` throwing a proper exception.

Reply:
This previous variant was `.map(this::readValue)`, line 288.

I was thinking more along the lines of the `rightPop(key, :Duration)` overloaded method above on line 262.

But, perhaps the choice of simply `.map(this:readRequiredValue)` vs. `.mapNotNull(this:readRequiredValue)` is due to the `right pop` followed by `left push` compound operation??